cmd/prometheus/main.go

```diff
@@ -401,6 +411,13 @@ func main() {
 		}
 	}
 
+	{ // overlapping blocks config
+
+		if !cfgFile.StorageConfig.TSDB.Retention.AllowOverlappingBlocks {
+			cfg.tsdb.AllowOverlappingBlocks = cfgFile.StorageConfig.TSDB.Retention.AllowOverlappingBlocks
+		}
```

Comment:
This means we only accept setting it to false? Look at my suggestion about making it a pointer, and then it will be 
```go
if cfgFile.StorageConfig.TSDB.AllowOverlappingBlocks != nil {
	cfg.tsdb.AllowOverlappingBlocks = *cfgFile.StorageConfig.TSDB.AllowOverlappingBlocks
}
```